elasticsearch/src/main/java/org/hibernate/search/elasticsearch/util/impl/DigestSelfSigningCapable.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.util.impl;
+
+import java.io.IOException;
+import java.security.MessageDigest;
+
+/**
+ * In some cases we can have more efficient code by delegating
+ * the computation of digests to the container object.
+ *
+ * @author Sanne Grinovero (C) 2017 Red Hat Inc.
+ */
+public interface DigestSelfSigningCapable {
```

Comment:
I've tried such things before. It didn't work out for various reasons.

1. the `writeTo(Outpustream)` is invoked in a different context by the http client and you can't do flow control in that case - not without buffering which is what I'm working to avoid.

> We could use a wrapper around the OutputStream to detect the content length if necessary.

That's too late. It's going to need the content lenght before the content is being produced, as the content lenght needs to be written in the output stream first thing, and also it might affect how the rest of the content is actually encoded (chunked, buffered, etc..).

Also you can't report a different lenght after you've produced the content as that gets you in trouble (bug of the client?).